File: lists_and_hashing/two_sum.py
class Solution(object):
    def two_sum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dictionary = {}
        for i, num in enumerate(nums):
            result = target - num
            if result in dictionary:
                return [dictionary[result], i]
            else:
                dictionary[num] = i
        return []
    
my_solution = Solution()
nums = [1, 2, 5, 7, 8]
target = 10
print(my_solution.two_sum(nums, target))

File: lists_and_hashing/twoSum.js
class Solution {
  twoSum(nums, target) {
    const hashmap = {};
    for (var i = 0; nums.length; i++) {
      const diff = target - nums[i];
      if (diff in hashmap) {
        return [hashmap[diff], i];
      } else {
        hashmap[nums[i]] = i;
      }
    }
    return [];
  }
}

const my_solution = new Solution();
nums = [1, 2, 5, 8];
target = 10;
console.log(my_solution.twoSum(nums, target));
